### ISSUE TITLE
HELP! Fatal error: 'Flutter/Flutter.h' file not found #import <Flutter/Flutter.h> - iOS

### ISSUE THREAD

**Issue by Alvaro-99** [flutter/flutter#84921]
Hello! My app was working well until I came across the following issue: "Fatal error: 'Flutter/Flutter.h' file not found #import <Flutter/Flutter.h>" :

```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
lib/main.dart:1
Xcode build done.                                            8.1s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    In file included from /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.m:5:
    /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    note: Removed stale file '/Users/XnameX/Library/Developer/Xcode/DerivedData/Runner-cfjslqyzzdpnoyeoygvffcmbgtpq/Build/Products/Debug-iphonesimulator/sqflite/sqflite.framework'
Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
Exited (sigterm)
```

What can I do to resolve this? Thanks a lot in advance!

**Comment by behzodfaiziev** [comment]
Can you provide `  flutter doctor -v`

**Comment by behzodfaiziev** [comment]
Similar issue have been discussed in this link: 
fatal error: 'Flutter\Flutter.h' file not found #import <Flutter\Flutter.h> - IOS #70895

**Comment by Alvaro-99** [comment]
Below is my flutter doctor -v. There does not seem to be any issues for iOS.
```
(base)XnameX-MacBook-Pro:projectX XnameX$ flutter -v doctor
[✓] Flutter (Channel master, 2.3.0-17.0.pre.411, on macOS 11.4 20F71 darwin-x64, locale en-US)
    • Flutter version 2.3.0-17.0.pre.411 at /Users/XnameX/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 25ba467b5a (2 days ago), 2021-06-18 21:24:02 -0400
    • Engine revision ced58ef9c1
    • Dart version 2.14.0 (build 2.14.0-217.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 1E3C6717-657E-4D6B-B476-B94E443A6420 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 91.0.4472.106

! Doctor found issues in 1 category.
```
Also, I have checked with similar questions on the internet, but I was not able to find a solution that worked. What solution do you propose?

**Comment by Alvaro-99** [comment]
Hello! I tried this link (https://www.kindacode.com/article/flutter-fatal-error-flutter-flutter-h-file-not-found/) and followed step-by-step the instructions in this link, but I still get the same error. What should I do?

**Comment by AkaankshK** [comment]
Just give this a try:

Upgrade all your dependencies such as Firebase etc. to the latest version.

Inside your ios folder, take a backup of the Runner folder(preferable whole ios folder). Then delete the ios folder.

Now in your project directory run "flutter create ." 

This will regenerate the ios folder. 

Now replace the new Runner folder with the one you've backed up.

Check your Podfile and try installing them. After successful installation of pods, try to run the app.

Hopefully this should fix.

**Comment by Alvaro-99** [comment]
Hello! Thanks for the help, but unfortunately is still does NOT work. I still get the following problem on my Debug Console:
```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
lib/main.dart:1
Xcode build done.                                            9.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    In file included from /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.m:5:
    /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
Exited (sigterm)
```
What should I do?

**Comment by Alvaro-99** [comment]
Then I tried running it after doing flutter clean and pod install, and I still had the same error. Do you what is the source of this problem?

**Comment by zmm2tysu** [comment]
check if you have import the package dart:html
I remove that package, then the error disappear

**Comment by Alvaro-99** [comment]
Hello, I  have checked but I have nothing like dart:html. The only thing I use once is "import 'dart:io';". Do you know what the source of the problem is? Thanks!

**Comment by Alvaro-99** [comment]
After investigating the problem, the issue seems to come from 2 docs: "FLTPathProviderPlugin.h” (issue: “'Flutter/Flutter.h' file not found”) and “FLTPathProviderPlugin.m” (issue: “1. In file included from /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.m:5:”). Something that I noticed is that the folder “.pub-cache” does NOT seem to be in Users/XnameX/Desktop/flutter when I look for it in my Finder. What could it be?

**Comment by aDrIaNo34500** [comment]
FR

**Comment by jmagman** [comment]
@AkaankshK  Please don't suggest workarounds like "regenerate your ios directory" in this bug tracker, it makes it impossible to actually understand the underlying bug so we can fix it for everyone.  That advice is so endemic that it makes it really hard for us to figure out what people are hitting.

**Comment by AkaankshK** [comment]
> Hello! Thanks for the help, but unfortunately is still does NOT work. I still get the following problem on my Debug Console:
> 
> ```
> Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
> lib/main.dart:1
> Xcode build done.                                            9.7s
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     ** BUILD FAILED **
> Xcode's output:
> ↳
>     In file included from /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.m:5:
>     /Users/XnameX/Desktop/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-2.0.2/ios/Classes/FLTPathProviderPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Analyzing workspace
>     note: Constructing build description
>     note: Build preparation complete
> Could not build the application for the simulator.
> Error launching application on iPhone 12 Pro Max.
> Exited (sigterm)
> ```
> 
> What should I do?

Sorry no idea. I get this error from time to time, this usually fixes it although its a very bad workaround.

**Comment by shoeb1000** [comment]
I am also having same problem

**Comment by TahaTesser** [comment]
Hi @Alvaro-99 
First of all, please don't execute `pod init`, `pod install` in your ios folder

Please existing Pod files and Podspec
Execute `flutter pub get`
then  `flutter run  -v` (this will generate all pod files for you (

If the issue problem persists, please provide your complete logs, paste them in a fie as logs for iOS exceed the GitHub comment size limit 
Thank you

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
